imageio/imageio-webp/src/main/java/com/twelvemonkeys/imageio/plugins/webp/lossless/VP8LDecoder.java

```diff
@@ -135,33 +135,44 @@ public void readVP8Lossless(final WritableRaster raster, final boolean topLevel,
         }
 
         if (fullSizeRaster != raster && param != null) {
-            // Copy into destination raster with settings applied
-            Rectangle sourceRegion = param.getSourceRegion();
-            int sourceXSubsampling = param.getSourceXSubsampling();
-            int sourceYSubsampling = param.getSourceYSubsampling();
-            int subsamplingXOffset = param.getSubsamplingXOffset();
-            int subsamplingYOffset = param.getSubsamplingYOffset();
-            Point destinationOffset = param.getDestinationOffset();
-
-            if (sourceRegion == null) {
-                sourceRegion = raster.getBounds();
-            }
+            copyIntoRasterWithParams(fullSizeRaster, raster, param);
+        }
+    }
+    
+    /**
+     * Copy a source raster into a destination raster with settings applied.
+     * 
+     * TODO: This piece of code was inside readVP8Lossless(), but as it can be used
+     * in readAlpha(), this static utility method was created here. Probably, it
+     * can be moved to another class.
+     */
+    public static void copyIntoRasterWithParams(final Raster srcRaster, final WritableRaster dstRaster,
+            final ImageReadParam param) {
+        Rectangle sourceRegion = param.getSourceRegion();
+        int sourceXSubsampling = param.getSourceXSubsampling();
+        int sourceYSubsampling = param.getSourceYSubsampling();
+        int subsamplingXOffset = param.getSubsamplingXOffset();
+        int subsamplingYOffset = param.getSubsamplingYOffset();
+        Point destinationOffset = param.getDestinationOffset();
+
+        if (sourceRegion == null) {
+            sourceRegion = dstRaster.getBounds();
+        }
```

Comment:
`param` null safe fix:

```suggestion
        Rectangle sourceRegion = param != null && param.getSourceRegion() != null ? param.getSourceRegion() : dstRaster.getBounds();
        int sourceXSubsampling = param != null ? param.getSourceXSubsampling() : 1;
        int sourceYSubsampling = param != null ? param.getSourceYSubsampling() : 1;
        int subsamplingXOffset = param != null ? param.getSubsamplingXOffset() : 0;
        int subsamplingYOffset = param != null ? param.getSubsamplingYOffset() : 0;
        Point destinationOffset = param != null ? param.getDestinationOffset() : new Point(0, 0) ;
```
This should make the `param == null` case fall into the fast case using `setRect` so no big performance penalty.